src/mrinufft/operators/interfaces/nudft_numpy.py

```diff
@@ -4,23 +4,42 @@
 
 import numpy as np
 import scipy as sp
-
 from ..base import FourierOperatorCPU
-from mrinufft._utils import proper_trajectory
+from mrinufft._utils import proper_trajectory, get_array_module
 
 
 def get_fourier_matrix(ktraj, shape, dtype=np.complex64, normalize=False):
     """Get the NDFT Fourier Matrix."""
+    module = get_array_module(ktraj)
     ktraj = proper_trajectory(ktraj, normalize="unit")
     n = np.prod(shape)
     ndim = len(shape)
-    matrix = np.zeros((len(ktraj), n), dtype=dtype)
-    r = [np.linspace(-s / 2, s / 2 - 1, s) for s in shape]
-    grid_r = np.reshape(np.meshgrid(*r, indexing="ij"), (ndim, np.prod(shape)))
-    traj_grid = ktraj @ grid_r
-    matrix = np.exp(-2j * np.pi * traj_grid, dtype=dtype)
+
+    if module.__name__ == "torch":
+        torch = module
+        device = ktraj.device
+        dtype = torch.complex64
+        r = [torch.linspace(-s / 2, s / 2 - 1, s, device=device) for s in shape]
+        grid_r = torch.meshgrid(r, indexing="ij")
+        grid_r = torch.reshape(torch.stack(grid_r), (ndim, n)).to(device)
+        traj_grid = torch.matmul(ktraj, grid_r)
+        matrix = torch.exp(-2j * np.pi * traj_grid).to(dtype).to(device).clone()
+    else:
+        r = [np.linspace(-s / 2, s / 2 - 1, s) for s in shape]
+        grid_r = np.reshape(np.meshgrid(*r, indexing="ij"), (ndim, np.prod(shape)))
+        traj_grid = ktraj @ grid_r
+        matrix = np.exp(-2j * np.pi * traj_grid, dtype=dtype)
+
     if normalize:
-        matrix /= np.sqrt(np.prod(shape)) * np.power(np.sqrt(2), len(shape))
+        matrix /= (
+            (
+                torch.sqrt(torch.tensor(np.prod(shape), device=device))
+                * torch.pow(torch.sqrt(torch.tensor(2, device=device)), ndim)
+            )
+            if module.__name__ == "torch"
+            else (np.sqrt(np.prod(shape)) * np.power(np.sqrt(2), len(shape)))
+        )
```

Comment:
This could potenitally be a problem if someone uses `normalize=True` with numpy! Try to follow using module here to prevent that